ir/src/constraints/mod.rs

```diff
@@ -93,6 +94,9 @@ pub(super) struct Constraints {
     /// vector or a matrix, a new root is added with a key equal to the [VariableValue] of the
     /// element.
     variable_roots: VariableRoots,
+
+    /// A set of all boundaries which have been constrained.
```

Comment:
nit: I'd add a sentence explaining what it is being used for (otherwise it is not immediately clear why we need this in addition to `boundary_constraints`).

Reply:
Updated!